packages/upset/src/components/AltTextSidebar.tsx

```diff
@@ -1,30 +1,59 @@
 import {
-  Divider, Drawer, IconButton, TextField, Typography, css,
+  Box,
+  Divider,
+  Drawer,
+  FormControl,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+  Typography,
+  css,
 } from '@mui/material';
 import CloseIcon from '@mui/icons-material/Close';
-import { useState, useEffect } from 'react';
+import { useState, useEffect, FC, useContext } from 'react';
+import { ProvenanceContext } from './Root';
 
 type Props = {
-  open: boolean,
-  close: () => void,
-  generateAltText: () => Promise<string>,
+  open: boolean;
+  close: () => void;
+  generateAltText: (verbosity: string, explain: string) => Promise<string>;
 }
 
 const initialDrawerWidth = 450;
 
-export const AltTextSidebar = ({ open, close, generateAltText }: Props) => {
+export const AltTextSidebar: FC<Props> = ({ open, close, generateAltText }) => {
   const [altText, setAltText] = useState<string>('');
+  const [verbosity, setVerbosity] = useState<string>('low');
+  const [explain, setExplain] = useState<string>('full');
 
-  useEffect(() => {
-    async function generate() {
-      const resp = await generateAltText();
+  const { provenance } = useContext(ProvenanceContext);
+
+  function genAltText(): void {
+    async function generate(): Promise<void> {
+      const resp = await generateAltText(verbosity, explain);
       setAltText(resp);
     }
 
-    if (open) {
-      generate();
-    }
-  }, [open]);
+    generate();
+  }
+
+  useEffect(() => {
+    genAltText();
+  }, [verbosity, explain]);
+
+  provenance.currentChange(() => {
+    genAltText();
+  });
```

Comment:
Is there no way we can put something from provenance in the dependencies on the use effect so we don't need 2 ways to call it. This isn't bad, at all, it just might be possible to simplify it